### PR TITLE
feat(form): Filter parameters in the Canvas Form

### LINKED ISSUE #701: Filter parameters in the Canvas Form
### Please describe the feature that you want to propose

Looking through many parameters is cumbersome, it would be beneficial to have a filter in the Canvas Form that allows one to look for a given parameter.

### PR DESCRIPTION
**Pull request by shivamG640**
### Description
Added Filter parameters feature in the Canvas Form. Fixes #701 
Check here: https://shivamg640.github.io/kaoto/

### CODE REVIEW

**Comment by lhein** [comment]
looks great. One thing we could improve....when for instance filtering an AMQP component and search for "password" I only get a collapsed section that I manually need to expand. Would be good to expand all sections in the results when filtering if possible

File: packages/ui/src/components/Form/customField/CustomNestField.tsx [hunk 1]
```diff
@@ -43,9 +44,16 @@ export const CustomNestField = connectField(
...
+      field[0].includes(filteredFieldText),
+    );
+    const actualProperties = Object.fromEntries(filteredProperties);
     const propertiesArray = useMemo(() => {
-      return getFieldGroups(props.properties ?? {});
-    }, [props.properties]);
+      return getFieldGroups(actualProperties);
+    }, [actualProperties]);
+
+    if (propertiesArray.common.length === 0 && Object.keys(propertiesArray.groups).length === 0) return <></>;
```
Comment (lordrip) [inline packages/ui/src/components/Form/customField/CustomNestField.tsx:56]: Maybe we can return `null` here, so react doesn't render anything

**Comment by shivamG640** [comment]
@lordrip , If tests for the newly added provider can go later, this PR is ready for the final review!

File: packages/ui/src/components/Form/customField/CustomNestField.tsx [hunk 2]
```diff
@@ -58,18 +68,13 @@ export const CustomNestField = connectField(
         {Object.entries(propertiesArray.groups)
           .sort((a, b) => (a[0] === 'advanced' ? 1 : b[0] === 'advanced' ? -1 : 0))
           .map(([groupName, groupFields]) => (
-            <ExpandableSection
-              key={`${groupName}-section-toggle`}
-              toggleText={capitalize(`${groupName} properties`)}
-              toggleId={`${groupName}-expandable-section-toggle`}
-              contentId="expandable-section-content"
-            >
+            <CustomExpandableSection groupName={groupName} isGroupExpanded={isGroupExpanded}>
```
Comment (lordrip) [inline packages/ui/src/components/Form/customField/CustomNestField.tsx:71]:
```suggestion
            <CustomExpandableSection key={`processor-${groupName}-section-toggle`} groupName={groupName} isGroupExpanded={isGroupExpanded}>
```

File: packages/ui/src/providers/filtered-field.provider.test.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+const TestComponent = () => {
+  const { filteredFieldText, onFilterChange, isGroupExpanded } = useContext(FilteredFieldContext)!;
+
+  return (
+    <div>
+      <input
+        type="text"
+        value={filteredFieldText}
+        onChange={(_event) => onFilterChange(null, _event.target.value)}
```
Comment (lordrip) [inline packages/ui/src/providers/filtered-field.provider.test.tsx:44]: [:nail_care:] We use the `_` prefix when the parameter is not used, in this case, `_event` is used.
```suggestion
        onChange={(event) => onFilterChange(null, event.target.value)}
```

File: packages/ui/src/providers/filtered-field.provider.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    },
+    [setSearchTerm],
+  );
+
+  return (
+    <FilteredFieldContext.Provider
+      value={{
+        filteredFieldText: searchTerm,
+        onFilterChange: onFilterChange,
+        isGroupExpanded: searchTerm.length > 0 ? true : false,
```
Comment (lordrip) [inline packages/ui/src/providers/filtered-field.provider.tsx:32]:
```suggestion
        isGroupExpanded: searchTerm.length > 0,
```

File: packages/ui/src/providers/filtered-field.provider.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+      setSearchTerm(value);
+    },
+    [setSearchTerm],
+  );
+
+  return (
+    <FilteredFieldContext.Provider
+      value={{
+        filteredFieldText: searchTerm,
+        onFilterChange: onFilterChange,
```
Comment (lordrip) [inline packages/ui/src/providers/filtered-field.provider.tsx:31]:
```suggestion
        onFilterChange,
```

**Review by lordrip — APPROVED** [review]
Looks good, thanks for implementing this.

**Pull request merged**